music_assistant/server/providers/siriusxm/__init__.py

```diff
@@ -223,7 +227,27 @@ async def browse(self, path: str) -> Sequence[MediaItemType | ItemMapping]:
         return [self._parse_radio(channel) for channel in self._channels]
 
     def _channel_updated(self, live_channel_raw: dict[str, Any]) -> None:
-        self.logger.debug(f"channel updated {live_channel_raw}")
+        """Handle a channel update event."""
+        live_data = XMLiveChannel.from_dict(live_channel_raw)
+
+        self.logger.debug(f"Got update for SiriusXM channel {live_data.id}")
+        current_channel = self._stream_details.item_id
+
+        if live_data.id != current_channel:
+            # This can happen when changing channels
+            self.logger.debug(
+                f"Received update for channel {live_data.id}, current channel is {current_channel}"
+            )
+            return
```

Comment:
Also, this keeps updating forever, even when the player switches to another source ?
Where is this callback being registered/subscribed ?

Reply:
No - the only time I've seen it is basically if you change the channel right as an update is coming through.  That can lead to a song title from the prior station being shown on the new station.

The callback is coming from the `sxm` library - I haven't dug into it too much but it's just grabbing the details when the playlist is being refreshed.  I _think_ by default it refreshes it every 30s.  If it starts refreshing a playlist, and the user changes to another station, it'll publish the update into the callback, but the user might be on another station at that point.

I'm not sure I want to clear out the `_stream_details` because it generally means it's reflecting a new station, rather than just lingering around forever, no?

Reply:
The callback is registered with the creation of the client: https://github.com/music-assistant/server/pull/1739/files#diff-cca71fa41dedc871088abbae063e6fb30781c8993798ad1547911ac6844e9531R137

Reply:
Simple test: stop playback and see if that callback is never called again

Reply:
My understanding is that this is called anytime the client refreshes the playlist with the provider.  When MA requests the m3u8 url, the sxm client will check to see if it's been 30s (by default, the HLS stream can override this depending on parameters) since it has last fetched the details, and if it's been that long, it'll refresh the playlist and trigger the callback with the live data.

If nothing is streaming the playlist from the client, it won't refresh the live data.

My guess is that there's a race condition where MA requests the updated playlist, the user changes channel, and then the client updates the original channels live data after fetching it over the network.

I can confirm this with the test you mentioned.

Reply:
OK thanks for confirming!